public/app/plugins/panel/state-timeline/StateTimelinePanel.tsx

```diff
@@ -1,19 +1,22 @@
-import React, { useCallback, useMemo } from 'react';
+import React, { useCallback, useMemo, useRef, useState } from 'react';
 
-import { DataFrame, FieldType, PanelProps } from '@grafana/data';
-import { TooltipPlugin, useTheme2, ZoomPlugin, usePanelContext } from '@grafana/ui';
+import { CartesianCoords2D, DataFrame, FieldType, PanelProps } from '@grafana/data';
+import { Portal, UPlotConfigBuilder, usePanelContext, useTheme2, VizTooltipContainer, ZoomPlugin } from '@grafana/ui';
+import { CloseButton } from 'app/core/components/CloseButton/CloseButton';
 import { getLastStreamingDataFramePacket } from 'app/features/live/data/StreamingDataFrame';
 
+import { HoverEvent, setupConfig } from '../barchart/config';
```

Comment:
I think the shared components will need to be moved somewhere else since they are used across multiple panels. My guess would be in `@grafana/data` and imported in all modified panels.